### PR TITLE
Rename bricks to databricks

### PR DESCRIPTION
**Pull request by fjakobs**
## Changes
Rename CLI from `bricks` to `databricks`

## Testing
- manually tested AWS OAuth
- manually tested sync

### CODE REVIEW

File: packages/databricks-sdk-js/src/config/Config.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export type AuthType =
     | "azure-cli"
     | "google-id"
     | "metadata-service"
-    | "bricks-cli";
```
Comment (fjakobs) [inline packages/databricks-sdk-js/src/config/Config.ts:64]: @nfx @pietern we need to make this change consistently. How did you name it in go?

**Review by pietern — APPROVED** [review]
Looks straightforward

File: packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export class BricksCliCredentials implements CredentialProvider {
                     }
                 } catch (e) {
                     throw new ConfigError(
-                        `bricks-cli: cannot unmarshal Bricks CLI result: ${e}`,
+                        `databricks-cli: cannot unmarshal Bricks CLI result: ${e}`,
```
[the remark this answers is not in the document]
Comment (fjakobs) [reply packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts:76]: done

File: packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export class BricksCliCredentials implements CredentialProvider {
             await ts();
         } catch (e) {
             if (e instanceof ConfigError) {
-                if (e.message.indexOf("Can't find 'bricks' command") >= 0) {
+                if (e.message.indexOf("Can't find 'databricks' command") >= 0) {
                     config.logger.debug(
                         "Most likely Bricks CLI is not installed"
```
[the remark this answers is not in the document]
Comment (fjakobs) [reply packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts]: done

File: packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pietern) [reply packages/databricks-sdk-js/src/config/DatabricksCliCredentials.ts:76]: Still remaining in message.

File: packages/databricks-sdk-js/src/config/Config.ts (hunk 1 above)
Comment (pietern) [reply packages/databricks-sdk-js/src/config/Config.ts:64]: Can you revert just the auth type name? Then we unblock merging this and we do it in a patch bump.

File: packages/databricks-vscode/src/cli/CliWrapper.test.ts [hunk 4]
```diff
@@ -19,9 +19,9 @@ const execFile = promisify(execFileCb);
 
 describe(__filename, () => {
     it("should embed a working bricks CLI", async () => {
-        const bricksPath = __dirname + "/../../bin/bricks";
+        const bricksPath = __dirname + "/../../bin/databricks";
```
Comment (pietern) [inline packages/databricks-vscode/src/cli/CliWrapper.test.ts:22]: `cliPath`

File: packages/databricks-vscode/src/configuration/auth/AuthProvider.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ import {AzureCliCheck} from "./AzureCliCheck";
 import {BricksCliCheck} from "./BricksCliCheck";
 
 // TODO: Resolve this with SDK's AuthType.
-export type AuthType = "azure-cli" | "google-id" | "bricks-cli" | "profile";
+export type AuthType = "azure-cli" | "google-id" | "databricks-cli" | "profile";
```
Comment (pietern) [inline packages/databricks-vscode/src/configuration/auth/AuthProvider.ts:18]: Auth type revert, same as the other comment.

File: packages/databricks-vscode/src/configuration/auth/AuthProvider.ts [hunk 6]
```diff
@@ -142,23 +142,23 @@ export class BricksCliAuthProvider extends AuthProvider {
...
+            databricksPath: this.databricksPath,
         };
     }
 
     getSdkConfig(): Config {
         return new Config({
             host: this.host.toString(),
-            authType: "bricks-cli",
-            bricksCliPath: this.bricksPath,
+            authType: "databricks-cli",
```
Comment (pietern) [inline packages/databricks-vscode/src/configuration/auth/AuthProvider.ts:152]: Auth type

File: packages/databricks-vscode/src/configuration/auth/BricksCliCheck.ts [hunk 7]
```diff
@@ -73,8 +73,8 @@ export class BricksCliCheck implements Disposable {
         const workspaceClient = new WorkspaceClient(
             {
                 host: this.authProvider.host.toString(),
-                authType: "bricks-cli",
-                bricksCliPath: this.authProvider.bricksPath,
+                authType: "databricks-cli",
```
Comment (pietern) [inline packages/databricks-vscode/src/configuration/auth/BricksCliCheck.ts:76]: Auth type

File: packages/databricks-vscode/src/configuration/configureWorkspaceWizard.ts [hunk 8]
```diff
@@ -232,7 +232,7 @@ function authMethodsForHostname(host: URL): Array<AuthType> {
     }
 
     if (isAwsHost(host)) {
-        return ["bricks-cli", "profile"];
+        return ["databricks-cli", "profile"];
     }
 
     return ["profile"];
```
Comment (pietern) [inline packages/databricks-vscode/src/configuration/configureWorkspaceWizard.ts:238]: This file has a couple of auth type changes.

**Review by pietern — COMMENTED** [review]
Looking at all the auth type changes in here, let's instead get an updated SDK cut out and fix it there and avoid a separate minor bump.

**Comment by fjakobs** [comment]
@pietern we also need to rename these env vars:

BRICKS_ROOT
BRICKS_UPSTREAM
BRICKS_UPSTREAM_VERSION

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**